llvm/lib/Transforms/IPO/SampleProfile.cpp

```diff
@@ -2326,6 +2327,9 @@ PreservedAnalyses SampleProfileLoaderPass::run(Module &M,
 
   if (!FS)
     FS = vfs::getRealFileSystem();
+  if (!DisableSampleLoaderInlining.getNumOccurrences() &&
+      DisableSampleProfileInlining)
+    DisableSampleLoaderInlining = true;
```

Comment:
I think this mixes up global variable and class field and may be good to allow setting true and false from command line?
```suggestion
  if (!::DisableSampleLoaderInlining.getNumOccurrences())
    this->DisableSampleLoaderInlining = ::DisableSampleLoaderInlining;
```

Reply:
oh I think I misread the variables names. This actually changes the `cl::opt` setting based on the class parameter... I think that is dangerous then. In a ThinLTO or similar setting we could have multiple threads / LLVMContexts using different settings so we shouldn't just change the global variables IMO

Reply:
Ah, good point, I was to avoid setting the global variable in `PassBuilderPipelines.cpp`, so just learned this can also happen in the pass class. OK, I will try setting a class field.